### PR TITLE
Display tags in alphabetical order.

### LINKED ISSUE #240: Display tags in alphabetical order.
**Is your feature request related to a problem? Please describe.**
The `/tags` command currently displays tags in an order that is not alphabetical, maybe random, maybe based on date added.
This makes it difficult for users to find a tag.
**Describe the solution you'd like**
We display the tags in an alphabetical order.

**Additional context**
image

### PR DESCRIPTION
**Pull request by java-coding-prodigy**
Fixes #240

### CODE REVIEW

**Comment by java-coding-prodigy** [comment]
image
A sample output. The tags are in alphabetical order always

**Comment by Zabuzard** [comment]
Please check back the linked GH issue and the proposed change

File: application/src/main/java/org/togetherjava/tjbot/commands/tags/TagsCommand.java [hunk 1]
```diff
@@ -42,8 +45,17 @@ public TagsCommand(TagSystem tagSystem) {
     @Override
     public void onSlashCommand(@NotNull SlashCommandEvent event) {
         // TODO A list might be better than comma separated, which is hard to read
-        event.replyEmbeds(MessageUtils.generateEmbed("All available tags",
-                String.join(", ", tagSystem.getAllIds()), event.getUser(), TagSystem.AMBIENT_COLOR))
+        ArrayList<String> list = new ArrayList<>(tagSystem.getAllIds());
```
Comment (Heatmanofurioso) [inline application/src/main/java/org/togetherjava/tjbot/commands/tags/TagsCommand.java:48]: This list variable should use the interface List and not ArrayList

File: application/src/main/java/org/togetherjava/tjbot/commands/tags/TagsCommand.java [hunk 2]
```diff
@@ -42,8 +45,17 @@ public TagsCommand(TagSystem tagSystem) {
...
+
+        event
+            .replyEmbeds(MessageUtils.generateEmbed("All available tags",
```
Comment (Heatmanofurioso) [inline application/src/main/java/org/togetherjava/tjbot/commands/tags/TagsCommand.java:51]: You should add a simple test to ensure that this logic works as desired

**Review by Heatmanofurioso — CHANGES_REQUESTED** [review]

File: application/src/main/java/org/togetherjava/tjbot/commands/tags/TagsCommand.java (hunk 1 above)
Comment (java-coding-prodigy) [reply application/src/main/java/org/togetherjava/tjbot/commands/tags/TagsCommand.java:48]: The `List` interface does not have a `subList` method

Comment (Heatmanofurioso) [reply application/src/main/java/org/togetherjava/tjbot/commands/tags/TagsCommand.java:48]: I'm pretty sure it does
https://docs.oracle.com/javase/8/docs/api/java/util/List.html#subList-int-int-

File: application/src/main/java/org/togetherjava/tjbot/commands/tags/TagsCommand.java [hunk 3]
```diff
@@ -42,8 +45,17 @@ public TagsCommand(TagSystem tagSystem) {
     @Override
     public void onSlashCommand(@NotNull SlashCommandEvent event) {
         // TODO A list might be better than comma separated, which is hard to read
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/commands/tags/TagsCommand.java:47]: You should also remove the TODO that corresponds to what ur implementing here 😆

**Review by Zabuzard — CHANGES_REQUESTED** [review]

**Comment by Zabuzard** [comment]
Whats up with this PR? Will you continue on this or should someone else take over, if desired? Its a "good first issue" and others might want to take it over maybe :)

**Comment by Zabuzard** [comment]
Will close and reopen this for someone else in 3 days if no response. @java-coding-prodigy

**Comment by java-coding-prodigy** [comment]
I think we should discuss in the issue or discord what we are to do with this. The last time we chatted about this you said something about tabulating the tags in rows and columns.

**Comment by Zabuzard** [comment]
> I think we should discuss in the issue or discord what we are to do with this. The last time we chatted about this you said something about tabulating the tags in rows and columns.

i told u to investigate and tell me the max line limit for messages with content `* foobar`. like how often u can spam
```
* foobar
* foobar
* foobar
...
```
until its unhappy.

u havent reported back since then.

**Comment by Zabuzard** [comment]
@java-coding-prodigy So you continue or not?

File: application/src/main/java/org/togetherjava/tjbot/commands/tags/TagSystem.java [hunk 4]
```diff
@@ -140,14 +139,16 @@ void putTag(String id, String content) {
...
             try (var select = context.select(Tags.TAGS.ID)) {
                 return select.from(Tags.TAGS)
                     .fetch()
                     .stream()
                     .map(dbRecord -> dbRecord.getValue(Tags.TAGS.ID))
-                    .collect(Collectors.toSet());
+                    .distinct()
+                    .sorted()
+                    .toList();
             }
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/commands/tags/TagSystem.java:152]: From a design-pov, the sorting should not be done here but at the point where it is actually used. The system should still return a `Set<String>` and not a sorted distinct list.

File: application/src/main/java/org/togetherjava/tjbot/commands/tags/TagSystem.java [hunk 5]
```diff
@@ -140,14 +139,16 @@ void putTag(String id, String content) {
      *
      * @return a set of all ids known to the system, not backed
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/commands/tags/TagSystem.java:140]: The documention is outdated with the code change you propose. It is not a set anymore and its also sorted, which is not mentioned here at all (hence should not be relied on!)

**Review by Zabuzard — CHANGES_REQUESTED** [review]

File: application/src/main/java/org/togetherjava/tjbot/commands/tags/TagsCommand.java [hunk 6]
```diff
@@ -42,8 +45,17 @@ public TagsCommand(TagSystem tagSystem) {
...
+        ArrayList<String> list = new ArrayList<>(tagSystem.getAllIds());
+
+        event
+            .replyEmbeds(MessageUtils.generateEmbed("All available tags",
+                    IntStream.range(1, list.size())
+                        .filter(x -> x % 10 == 0 || x == list.size() - 1)
+                        .mapToObj(i -> String.join(", ",
+                                i % 10 != 0 && i != list.size() - 1 ? list.subList(i - 10, i)
+                                        : list.subList((i / 10) * 10, i + 1)))
+                        .collect(Collectors.joining("\n")),
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/commands/tags/TagsCommand.java:57]: What is this doing? A comment should clarify and explain it to readers.

Also, there arte a lot of "magic values" (in particular a lot of `10`s), what do they mean? Use some properly named constants instead.

Also think about extracting this logic into a helper method, if applicable.

**Review by Zabuzard — CHANGES_REQUESTED** [review]

**Comment by Zabuzard** [comment]
Closing this to make it available for others to work on again. No activity on this PR for a longer time and OP said someone else can take over.

This PR does not really have a lot of code that is worth to be rescued (in particular considering the design discussions and the CR comments).

**Pull request closed without merging**